lib/rules/no-duplicate-imports.js

```diff
@@ -21,18 +57,62 @@ function getValue(node) {
     return "";
 }
 
+
+/**
+ * Returns a boolean if we should report contradictory import type.
+ * @param {string[]} importTypes An array contain import types of a module.
+ * @param {string} importType An contradictory import type to check if we should report it or not.
+ * @returns {boolean} true if the contradictory import type should be reported.
+ */
+function shouldReportContradictoryImportType(importTypes, importType) {
+    if (importTypes.indexOf(importType) > -1) {
+        return true;
+    }
+    return (
+        importTypes.findIndex(importTypeItem => CONTRADICTORY_IMPORT_TYPES.includes(importTypeItem)) === -1
+    );
+
+}
```

Comment:
I think the third import shouldn’t be reported because the second one can be merged with the first one, so in this case, the third import can’t be merged with the first one!

So the code will be like this:
```

import mod, * as modns  from 'mod';

import { foo } from 'mod';

```
For your proposed logic, I think it will simplify the code.